File: cpp/SelectionSort.cpp
#include <iostream>
#include <iomanip>
#include <string>
#include <fstream>
#include <cmath>
#include <cstring>

using namespace std;
const int ARRAY_SIZE = 7;

void selectionSort(int list[], int length);

int main()
{
	int i;
	int arr[] = { 35, 12, 27, 18, 45, 16, 38 };

	cout << "The unsorted list is: ";

	for (i = 0; i < ARRAY_SIZE; i++) {
		cout << arr[i] << " ";
	}
	cout << endl;

	selectionSort(arr, ARRAY_SIZE);

	cout << "The sorted list is: ";

	for (i = 0; i < ARRAY_SIZE; i++) {
		cout << arr[i] << " ";
	}
	cout << endl;

	return 0;
}

void selectionSort(int list[], int length) {
	int index;
	int smallestIndex;
	int location;
	int temp;

	for (index = 0; index < length - 1; index++) {
		// step a (find smallest element)
		smallestIndex = index;
		for (location = index + 1; location < length; location++) {
			if (list[location] < list[smallestIndex]) {
				smallestIndex = location;
			}
		}
		// step b (swap elements)
		temp = list[smallestIndex];
		list[smallestIndex] = list[index];
		list[index] = temp;
	}
}

File: Java/Selection_sort.java
import java.util.*;

public class Selection_sort {
    
  public static void selectionSort(int[] arr) {
    //write your ode here
    for(int i =0;i<arr.length-1;i++){
        int mi =i;
        for(int j =i+1;j<arr.length;j++){
            if(isSmaller(arr,j,mi)){mi=j;}
        }
        swap(arr,i,mi);
    }
    
  }

  // used for swapping ith and jth elements of array
  public static void swap(int[] arr, int i, int j) {
   
    int temp = arr[i];
    arr[i] = arr[j];
    arr[j] = temp;
  }

  // return true if ith element is smaller than jth element
  public static boolean isSmaller(int[] arr, int i, int j) {

    if (arr[i] < arr[j]) {
      return true;
    } else {
      return false;
    }
  }

  public static void print(int[] arr) {
    for (int i = 0; i < arr.length; i++) {
      System.out.println(arr[i]);
    }
  }

  public static void main(String[] args) throws Exception {
    Scanner scn = new Scanner(System.in);
    int n = scn.nextInt();
    int[] arr = new int[n];
    for (int i = 0; i < n; i++) {
      arr[i] = scn.nextInt();
    }
    scn.close();
    selectionSort(arr);
    print(arr);
  }

}